File: gaia-sdk-python/gaia_sdk/graphql/request/input/ConnectRemoveNodeImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class ConnectRemoveNodeImpulse():
    """
    The specification to remove an edge of a certain type with a certain target from a node
    """
    type: EdgeType
    target: str

    def __init__(self, type: EdgeType, target: str):
        self.type = type
        self.target = target

    def __eq__(self, other):
        if type(other) is type(self):
            return self.type == other.type and self.target == other.target
        return False

    def __repr__(self):
        return {'type': self.type, 'target': self.target}

File: gaia-sdk-javascript/src/graphql/request/input/ConnectRemoveNodeImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * The specification to remove an edge of a certain type with a certain target from a node
 */
export class ConnectRemoveNodeImpulse {
public _typeName = "ConnectRemoveNodeImpulse";
    private type:EdgeType;
    private target:Uuid;

    constructor (type:EdgeType, target:Uuid) {
        this.type = type;
        this.target = target;
    }
}
